app/services/panoptes_api.rb

```diff
@@ -30,14 +37,25 @@ def project(slug)
     { id: response['id'].to_i, slug: response['slug'] }
   end
 
+  def workflow(id, include_project=nil)
+    query = { id: id }
+    response = if include_project
```

Comment:
Not blocking:

Using the `include_project` json api feature can result in a much slower API response as it the project resource has to load all the count and workflow data to serialize properly (especially as the project resource gathers workflows over time). Using the project `?cards=true` serializer context would have sped up the project fetch as well

Noting that you throw most of the workflow & Project data away, Noting the existence of client methods to request the workflow https://github.com/zooniverse/panoptes-client.rb/blob/6079789b1d5a7fffad55b667a350ca01e64d03bd/lib/panoptes/client/workflows.rb#L6

I might have approached it like you have and then request the project & workflow like you've done below. Noting you can [limit the fields returned on the workflow](https://github.com/zooniverse/Panoptes-Front-End/blob/3d8cba4edfcee7e4d8a27fd05495d14ac136811f/app/pages/admin/project-status.jsx#L89) and use the [projects cards query param](https://github.com/zooniverse/Panoptes-Front-End/blob/940a84f20f8f64f13a0ba659c1f29d8cb39405fd/app/lib/get-social-data.js#L7) to make both faster requests with more tailored data i'm using here.

Reply:
The point here is to make a single API call--get either the workflow alone, or the workflow with the included project. That's the diff between `get_workflow` and `get_workflow_with_project`, and the reason I have my own `#workflow` method. 

I can and surely will limit the workflow fields included--alls I need is the display name, anyway. But then the question is what's faster: two separate queries limited in scope, or one query that doesn't limit the serialized project? Answer is that it doesn't matter a ton, but the single call/include is a little faster:

```
time_one = Benchmark.measure do
  wf_get = api.get('workflows', {id: 2660, fields: 'display_name' } )
  p_get = api.get('projects', {id: 1663, cards: true})
end
=> #<Benchmark::Tms:0x00007fe864a52bf8 @cstime=0.0, @cutime=0.0, @label="", @real=0.28667100006714463, @stime=0.005006999999999984, @total=0.012298999999999949, @utime=0.007291999999999965>
time_two = Benchmark.measure { wf_p_get = api.get('workflows', {id: 2660, include: 'project'}) }
=> #<Benchmark::Tms:0x00007fe85e546278 @cstime=0.0, @cutime=0.0, @label="", @real=0.20306200021877885, @stime=0.005897999999999959, @total=0.02594199999999991, @utime=0.02004399999999995>

time_one.real
=> 0.33838400011882186
time_two.real
=> 0.2026570001617074

10x
time_one = Benchmark.measure do
  10.times do
    wf_get = api.get('workflows', {id: 2660, fields: 'display_name' } )
    p_get = api.get('projects', {id: 1663, cards: true})
  end
end
=> #<Benchmark::Tms:0x00007fe860df5548 @cstime=0.0, @cutime=0.0, @label="", @real=2.9692969999741763, @stime=0.03373599999999993, @total=0.10932499999999995, @utime=0.07558900000000002>
time_two = Benchmark.measure { 10.times { wf_p_get = api.get('workflows', {id: 2660, include: 'project'}) } }
=> #<Benchmark::Tms:0x00007fe86480a800 @cstime=0.0, @cutime=0.0, @label="", @real=2.402155000017956, @stime=0.021334000000000075, @total=0.0662870000000001, @utime=0.04495300000000002>

time_one.real
=> 2.9692969999741763
time_two.real
=> 2.402155000017956
```